### DISCUSSION TITLE
Do the samples in PSRAM remain ofter power off?

### DISCUSSION THREAD

**Discussion by baesek** [marcel-licence/esp32_midi_sampler#79] [Q&A] [answered]
Hi, I am wondering if and how long the data uploaded to the PSRAM will stay there without any power supply. 
https://www.infineon.com/cms/de/product/memories/psram-pseudostatic-dram/
here it says its volatile memory... 
I am wondering because i am thinking about if the samples playback noise might come from data loss in the psram.

**Comment by marcel-licence** [comment, marked as answer]
Hi Paul,

I think the PSRAM is like any other RAM, only storing data when powered on. To store the data you can write it to LITTLEFS (to flash) or to the SD Card if connected and implemented.
I could imagine to implement a function which will store and restore its data. You will also need data from RAM (the information how many samples, and where the samples are stored).

Best regards,
Marcel

**Reply by baesek** [reply]
Hi! Oh i now realize that I mixed something up. I assumed that PSRAM is that special memory for the LittleFs where the samples stored in the data folder are uploaded to. But then it makes sense. If I understand right, the ESP32 used on the Audio Kit has 4MB of flash memory, so actually when not using an sd card the strange limitation of the PSRAM from 8MB to 4MB doesn't matter.

**Reply by marcel-licence** [reply]
The limitation of the PSRAM has another reason. It seems that there is something like to pages where Arduino is only capable of handling one single page resulting in only 4MB of PSRAM.
Regarding the flash you can select how it should be partitioned. You can deselect OTA to have more memory for LittleFS or your program code.
